### ISSUE TITLE
Type is not a Type. But is a Function which returns Type when type is recursive

### ISSUE THREAD

**Issue by ciricc** [Hookyns/tst-reflect#42]
I've tried this sample with recursive type. And i found that type is a function, not a `Type`.
I have last version of `tst-reflect` and `ts-node`.

```ts
import { getType } from 'tst-reflect';

type Settings = {
  users: User[];
};

type User = {
  friends: User[];
};

console.log(
  getType<Settings>()
    .getProperties()[0]
    .type.getTypeArguments()[0]
    .getProperties()[0]
    .type.isObjectLike()) // isObjectLike() is not a function
```

**Comment by Hookyns** [comment]
Hello @ciricc,
TY for the issue. 
This is a bug. I'll look into it.

**Comment by Hookyns** [comment]
Fixed.

Try new versions `tst-reflect@0.7.5` & `tst-reflect-transformer@0.9.10`

**Comment by ciricc** [comment]
Hello, @Hookyns . Thank you for reply. I've tried new version but it seems like now is just undefined, not a function.

**Comment by Hookyns** [comment]
Okay, I'll check it again; this must be something with the Array, cuz I wrote test with:
```typescript
type Node = {
	left: Node;
	right: Node;
};
```
it didn't work too and now it works. I should try your example first, sorry.

**Comment by Hookyns** [comment]
@ciricc
Fixed in tst-reflect@0.7.6

**Comment by ciricc** [comment]
@Hookyns Thank you very much! Now is working ok
